code/script.js

```diff
@@ -227,99 +258,166 @@ const setSecret = () => {
+      category: category,
+    }
+  } else if (category === 'other') {
+    currentQuestion = {
+      attribute: value,
+      value: true,
+      category: category,
+    }
   }
 }
 
-// This function should be invoked when you click on 'Find Out' button.
+// This is the function that happens once you click "find out"
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
-
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
 
-  }
+  if (secret[currentQuestion.attribute] === currentQuestion.value) { 
+    filterCharacters(true) //To see if they are the same 
+  } else {
+    filterCharacters(false) //Or not the same
+  }  
 }
 
-// It'll filter the characters array and redraw the game board.
+// This will filter the characters and remove after question
 const filterCharacters = (keep) => {
-  const { category, value } = currentQuestion
-  // Show the correct alert message for different categories
-  if (category === 'accessories') {
+  group = currentQuestion.category
+  console.log(keep) //false or true
+  console.log(group)// eg. haircolor, accessories
+  console.log(currentQuestion.category)
+  
+  // Show the correct alert messages for keep or remove
+  if (group === 'accessories') {
     if (keep) {
       alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
+        `Yes, the person wears ${currentQuestion.attribute}! Keep all that wears ${currentQuestion.attribute}`
       )
     } else {
       alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
+        `No, the person doesn't wear ${currentQuestion.attribute}! Remove all that wears ${currentQuestion.attribute}`
       )
     }
-  } else if (category === 'other') {
-    // Similar to the one above
-  } else {
+  } else if (group === 'eye color') {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert(
+        `Yes the person has ${currentQuestion.value} eyes! Keep all with eyes ${currentQuestion.value}`
+      )
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert (
+        `No the person has not ${currentQuestion.value} eyes! Remove all with eyes ${currentQuestion.value}`
+      )
+    }
+  } else if (group === 'other') {
+    if (keep) { //attribute to show glasses, smoker and eye patch
+      alert(
+        `Yes, the person is a  ${currentQuestion.attribute}! Keep all persons who are a ${currentQuestion.attribute}`
```

Comment:
This part was a little bit hard to follow when it's not structured (i don't know the word for it in english). Some of the code can be more to the right so it's easy to see which if-statement it belongs to. 

It continues down further also, but i couldn't select it all :P